src/nucleus/tt.ml

```diff
@@ -135,10 +133,12 @@ and instantiate es ?(lvl=0) ({term=e';assumptions;loc;} as e) =
   in
   match e' with
 
-    | Type -> {e with assumptions}
+    | Type -> {e with assumptions} (* XXX why do we need assumptions here? Type is well formed by itself *)
```

Comment:
It's not about being well-formed, it's about having the right type.
Consider the following:

    constant A : Type
    constant B : Type
    constant a : Type

    do assume eq : A == B in a : B
This produces a term `e = Tt.Constant "a"` with assumptions `eq`. If we instantiate `eq` with some other term (which has it's own assumptions) in `e`, we want to get a term which has type `B`, so it needs to have the assumptions of what we instantiate `eq` with.

When `e = Type` it's the same with the constants `A` and `a` both replaced with `Type`, but `B` is still an arbitrary term. Therefore the assumptions on `Type` aren't special.